aspnetcore/test/http-files.md

```diff
@@ -128,15 +103,334 @@ Accept-Language: en-US,en;q=0.5
+A variable defined in an environment file can be the same as one defined in the `.http` file, or it can be different. If a variable is defined in both the `.http` file and the environment file, the value in the `.http` file overrides the value in the environment file.
+
+## User-specific environment files
+
+A user-specific value is any value that an individual developer wants to test with but doesn’t want to share with the team. Since the `http-client.env.json` file is checked in to source control by default, it wouldn’t be appropriate to add user-specific values to this file. Instead, put them in a file named `http-client.env.json.user` located in the same folder as the `http-client.env.json` file. Files that end with `.user` should be excluded from source control by default when using Visual Studio source control features.
+
+When the `http-client.env.json` file is loaded, Visual Studio looks for a sibling `http-client.env.json.user` file. If a variable is defined in an environment in both the `http-client.env.json` file and the `http-client.env.json.user` file, the value in the `http-client.env.json.user` file wins.
+
+Here's an example scenario that shows how a user-specific environment file works. Suppose the `.http` file has the following content:
+
+```http
+GET {{HostAddress}}/{{Path}}
+Accept: application/json
+```
+
+And suppose the `http-client.env.json` file contains the following content:
+
+```json
+{
+  "dev": {
+    "HostAddress": "https://localhost:7128",
+    "Path": "/weatherforecast"
+  },
+  "remote": {
+    "HostAddress": "https://contoso.com",
+    "Path": "/weatherforecast"
+  }
+}
+```
+
+And suppose there's a user-specific environment file that contains the following content:
+
+```json
+{
+  "dev": {
+    "Path": "/swagger/index.html"
+  }
+}
+```
+
+When the user selects the "dev" environment, the request is sent to `https://localhost:7128/swagger/index.html` because the `Path` value in the `http-client.env.json.user` file overrides the value from the `http-client.env.json` file.
+
+With the same environment files, suppose the variables are defined in the `.http` file:
+
+```http
+@HostAddress=https://contoso.com
+@Path=/weatherforecast
+
+GET {{HostAddress}}/{{Path}}
+Accept: application/json
+```
+
+In this scenario, the "dev" environment request is sent to `https://contoso.com/weatherforecast` because variable definitions in `.http` files override environment file definitions.
+
+## ASP.NET Core user secrets
+
+To get a value from [user secrets](xref:security/app-secrets), use an environment file that is located in the same folder as the ASP.NET Core project. In the environment file, define a variable that has `provider` and `secretName` properties. Set the `provider` value to `AspnetUserSecrets` and set `secretName` to the name of the desired user secret. For example, the following environment file defines a variable named `TemplatesApiKeyDev` that gets its value from the `config:TemplatesApiKeyDev` user secret:
```

Comment:
Can we change `TemplatesApiKey` to just `ApiKey`, that would be better for the readers.